src/main/java/com/microsoft/spring/data/gremlin/repository/support/SimpleGremlinRepository.java

```diff
@@ -46,27 +56,24 @@ public SimpleGremlinRepository(GremlinEntityInformation<T, ID> information, @Non
 
     @Override
     public Iterable<T> findAll() {
-        if (this.information.isEntityVertex()) {
-            return this.operations.findAll(this.information.getJavaType());
-        } else if (this.information.isEntityEdge()) {
-            return this.operations.findAll(this.information.getJavaType());
+        final GremlinSource source = GremlinUtils.toGremlinSource(getDomainClass());
+
+        if (source instanceof GremlinSourceGraph) {
+            throw new UnsupportedOperationException("findAll of Graph is not supported");
         }
 
-        throw new UnsupportedOperationException("findAll of Graph is not supported");
+        return this.operations.findAll(getDomainClass());
     }
 
     @Override
     public List<T> findAllById(@NonNull Iterable<ID> ids) {
-        final List<T> results = new ArrayList<>();
-
-        ids.forEach(id -> this.findById(id).ifPresent(results::add));
-
-        return results;
+        return StreamSupport.stream(ids.spliterator(), true).map(this::findById)
```

Comment:
Is it really necessary to use a parallelStream here? By default, all calls to parallelStream will use the shared fork-join thread pool, and we could potentially run into some contention issues. 
Maybe the parallel behavior should be optional?